lib/vrl/compiler/src/context.rs

```diff
@@ -37,6 +44,10 @@ impl<'a> Context<'a> {
         &mut self.state
     }
 
+    pub fn get_enrichment_tables(&self) -> &Option<Box<dyn EnrichmentTableSearch>> {
+        self.enrichment_tables
+    }
```

Comment:
It's more idiomatic to return an owned option of a reference.

```suggestion
    pub fn get_enrichment_tables(&self) -> Option<&Box<dyn EnrichmentTableSearch>> {
        self.enrichment_tables.as_ref()
    }
```